<a href="https://colab.research.google.com/github/daksha200/ML-Projects/blob/testrepo/Cryptocurrency_Price_Prediction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

 It is a digital medium of exchange that is encrypted and decentralized. Many people use cryptocurrencies as a form of investing because it gives great returns even in a short period.

Buying and selling result in a change in the price of any cryptocurrency, but buying and selling trends depend on many factors. Using machine learning for cryptocurrency price prediction can only work in situations where prices change due to historical prices that people see before buying and selling their cryptocurrency. So, in the section below, I will take you through how you can predict the bitcoin prices (which is one of the most popular cryptocurrencies) for the next 30 days.

I’ll start the task of Cryptocurrency price prediction by importing the necessary Python libraries and the dataset we need. For this task, I will collect the latest Bitcoin prices data from Yahoo Finance, using the yfinance API. This will help you collect the latest data each time you run this code:

In [3]:
!pip install yfinance

     |████████████████████████████████| 6.4 MB 6.5 MB/s 
     |████████████████████████████████| 63 kB 1.6 MB/s 
  Attempting uninstall: requests
    Found existing installation: requests 2.23.0
    Uninstalling requests-2.23.0:
      Successfully uninstalled requests-2.23.0
  Attempting uninstall: lxml
    Found existing installation: lxml 4.2.6
    Uninstalling lxml-4.2.6:
      Successfully uninstalled lxml-4.2.6
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
google-colab 1.0.0 requires requests~=2.23.0, but you have requests 2.27.1 which is incompatible.
datascience 0.10.6 requires folium==0.2.1, but you have folium 0.8.3 which is incompatible.


In [7]:
import pandas as pd
import yfinance as yf
import datetime
from datetime import date, timedelta
today = date.today()

d1 = today.strftime("%Y-%m-%d")
end_date = d1
d2 = date.today() - timedelta(days= 730)
d2 = d2.strftime("%Y-%m-%d")
start_date = d2

data = yf.download('BTC-USD', start = start_date, end = end_date, progress = False)
data["Date"] = data.index
data = data[["Date", "Open", "High", "Low", "Close", "Adj Close", "Volume"]]
data.reset_index(drop=True, inplace=True)
print(data.head())

        Date         Open         High          Low        Close    Adj Close  \
0 2020-03-29  6245.624512  6250.467285  5920.085938  5922.042969  5922.042969   
1 2020-03-30  5925.538574  6517.195801  5903.234375  6429.841797  6429.841797   
2 2020-03-31  6430.606445  6504.515137  6374.162109  6438.644531  6438.644531   
3 2020-04-01  6437.319336  6612.573730  6202.373535  6606.776367  6606.776367   
4 2020-04-02  6606.776367  7088.247559  6595.918457  6793.624512  6793.624512   

        Volume  
0  28373690931  
1  37101651525  
2  32786468812  
3  40346426266  
4  47660646124  


In the above code, I have collected the latest data of Bitcoin prices for the past 730 days, and then I have prepared it for any data science task. Now, let’s have a look at the shape of this dataset to see if we are working with 730 rows or not:

In [9]:
data.shape

(731, 7)

So the dataset contains 731 rows, where the first row contains the names of each column. Now let’s visualize the change in bitcoin prices till today by using a candlestick chart:

In [11]:
import plotly.graph_objects as go
figure = go.Figure(data=[go.Candlestick(x=data["Date"],
                                        open=data["Open"], 
                                        high=data["High"],
                                        low=data["Low"], 
                                        close=data["Close"])])
figure.update_layout(title = "Bitcoin Price Analysis", 
                     xaxis_rangeslider_visible=False)
figure.show()

The Close column in the dataset contains the values we need to predict. So, let’s have a look at the correlation of all the columns in the data concerning the Close column:

In [13]:
correlation = data.corr()
print(correlation["Close"].sort_values(ascending=False))

Close        1.000000
Adj Close    1.000000
High         0.998634
Low          0.998353
Open         0.996834
Volume       0.285698
Name: Close, dtype: float64


Predicting the future prices of cryptocurrency is based on the problem of Time series analysis. The AutoTS library in Python is one of the best libraries for time series analysis. So here I will be using the AutoTS library to predict the bitcoin prices for the next 30 days:

In [17]:
!pip install AutoTS

     |████████████████████████████████| 460 kB 4.1 MB/s 


In [21]:
from autots import AutoTS

model = AutoTS(forecast_length=30, frequency='infer', ensemble='simple')
model = model.fit(data, date_col='Date', value_col='Close', id_col=None)
prediction = model.predict()
forecast = prediction.forecast

print(forecast)

Inferred frequency is: D
Model Number: 1 with model AverageValueNaive in generation 0 of 10
Model Number: 2 with model AverageValueNaive in generation 0 of 10
Model Number: 3 with model AverageValueNaive in generation 0 of 10
Model Number: 4 with model DatepartRegression in generation 0 of 10
Model Number: 5 with model DatepartRegression in generation 0 of 10


/usr/local/lib/python3.7/dist-packages/sklearn/svm/_base.py:1208: ConvergenceWarning:

Liblinear failed to converge, increase the number of iterations.



Model Number: 6 with model DatepartRegression in generation 0 of 10


/usr/local/lib/python3.7/dist-packages/sklearn/neural_network/_multilayer_perceptron.py:549: ConvergenceWarning:

lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html



Model Number: 7 with model DatepartRegression in generation 0 of 10
Epoch 1/50
22/22 [==============================] - 12s 10ms/step - loss: 0.4162
Epoch 2/50
22/22 [==============================] - 0s 9ms/step - loss: 0.4153
Epoch 3/50
22/22 [==============================] - 0s 10ms/step - loss: 0.4134
Epoch 4/50
22/22 [==============================] - 0s 14ms/step - loss: 0.4129
Epoch 5/50
22/22 [==============================] - 0s 12ms/step - loss: 0.4126
Epoch 6/50
22/22 [==============================] - 0s 14ms/step - loss: 0.4082
Epoch 7/50
22/22 [==============================] - 0s 12ms/step - loss: 0.4042
Epoch 8/50
22/22 [==============================] - 0s 11ms/step - loss: 0.4008
Epoch 9/50
22/22 [==============================] - 0s 11ms/step - loss: 0.3968
Epoch 10/50
22/22 [==============================] - 0s 10ms/step - loss: 0.3950
Epoch 11/50
22/22 [==============================] - 0s 14ms/step - loss: 0.3901
Epoch 12/50
22/22 [==============================]

/usr/local/lib/python3.7/dist-packages/sklearn/neural_network/_multilayer_perceptron.py:549: ConvergenceWarning:

lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html



Model Number: 43 with model GLS in generation 0 of 10
Model Number: 44 with model SeasonalNaive in generation 0 of 10
Model Number: 45 with model GLM in generation 0 of 10
Template Eval Error: TypeError("ufunc 'isfinite' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''") in model 45: GLM
Model Number: 46 with model ETS in generation 0 of 10
ETS error TypeError("__init__() got an unexpected keyword argument 'damped_trend'")


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 47 with model FBProphet in generation 0 of 10
Model Number: 48 with model GluonTS in generation 0 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in model 48: GluonTS
Model Number: 49 with model UnobservedComponents in generation 0 of 10
Template Eval Error: ValueError("regression_type='User' but no future_regressor supplied") in model 49: UnobservedComponents
Model Number: 50 with model VAR in generation 0 of 10
Template Eval Error: ValueError('Only gave one variable to VAR') in model 50: VAR
Model Number: 51 with model VECM in generation 0 of 10
Template Eval Error: ValueError('Only gave one variable to VECM') in model 51: VECM
Model Number: 52 with model WindowRegression in generation 0 of 10
Template Eval Error: ValueError("regression_type='User' but no future_regressor passed") in model 52: WindowRegression
Model Number: 53 with model DatepartRegression in generation 0 of 10
Model Number: 54 w

INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 100 with model GLS in generation 0 of 10
Model Number: 101 with model UnobservedComponents in generation 0 of 10
Template Eval Error: ValueError('Provided exogenous values are not of the appropriate shape. Required (30, 1), got (30,).') in model 101: UnobservedComponents
Model Number: 102 with model GluonTS in generation 0 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in model 102: GluonTS
Model Number: 103 with model SeasonalNaive in generation 0 of 10
Model Number: 104 with model UnobservedComponents in generation 0 of 10
Template Eval Error: ValueError("regression_type='User' but no future_regressor supplied") in model 104: UnobservedComponents
Model Number: 105 with model NVAR in generation 0 of 10
Model Number: 106 with model FBProphet in generation 0 of 10
Template Eval Error: ValueError("regression_type='User' but no future_regressor passed") in model 106: FBProphet
Model Number: 107 with 

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:57: RuntimeWarning:

divide by zero encountered in true_divide

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:58: RuntimeWarning:

divide by zero encountered in true_divide

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:58: RuntimeWarning:

invalid value encountered in true_divide



Model Number: 112 with model DatepartRegression in generation 0 of 10
Epoch 1/100
9/9 [==============================] - 8s 113ms/step - loss: 1241.4806 - val_loss: 1407.5394
Epoch 2/100
9/9 [==============================] - 0s 49ms/step - loss: 1241.6107 - val_loss: 1407.5295
Epoch 3/100
9/9 [==============================] - 0s 45ms/step - loss: 1241.4126 - val_loss: 1407.5145
Epoch 4/100
9/9 [==============================] - 0s 45ms/step - loss: 1241.4357 - val_loss: 1407.5005
Epoch 5/100
9/9 [==============================] - 0s 47ms/step - loss: 1241.4471 - val_loss: 1407.4910
Epoch 6/100
9/9 [==============================] - 0s 45ms/step - loss: 1241.2936 - val_loss: 1407.4813
Epoch 7/100
9/9 [==============================] - 0s 51ms/step - loss: 1241.3059 - val_loss: 1407.4680
Epoch 8/100
9/9 [==============================] - 0s 46ms/step - loss: 1241.1809 - val_loss: 1407.4548
Epoch 9/100
9/9 [==============================] - 0s 45ms/step - loss: 1241.3027 - val_loss: 140

INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 127 with model VAR in generation 0 of 10
Template Eval Error: ValueError('Only gave one variable to VAR') in model 127: VAR
Model Number: 128 with model GluonTS in generation 0 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in model 128: GluonTS
Model Number: 129 with model MultivariateMotif in generation 0 of 10
Model Number: 130 with model Theta in generation 0 of 10
Template Eval Error: ModuleNotFoundError("No module named 'statsmodels.tsa.forecasting'") in model 130: Theta
Model Number: 131 with model VAR in generation 0 of 10
Template Eval Error: ValueError('Only gave one variable to VAR') in model 131: VAR
Model Number: 132 with model NVAR in generation 0 of 10
Model Number: 133 with model Theta in generation 0 of 10
Template Eval Error: ModuleNotFoundError("No module named 'statsmodels.tsa.forecasting'") in model 133: Theta
Model Number: 134 with model DatepartRegression in generation 0 of 

/usr/local/lib/python3.7/dist-packages/sklearn/svm/_base.py:1208: ConvergenceWarning:

Liblinear failed to converge, increase the number of iterations.

/usr/local/lib/python3.7/dist-packages/sklearn/svm/_base.py:1208: ConvergenceWarning:

Liblinear failed to converge, increase the number of iterations.



Model Number: 187 with model DatepartRegression in generation 1 of 10
Template Eval Error: ValueError('Failed to convert a NumPy array to a Tensor (Unsupported object type int).') in model 187: DatepartRegression
Model Number: 188 with model UnobservedComponents in generation 1 of 10
Template Eval Error: ValueError("regression_type='User' but no future_regressor supplied") in model 188: UnobservedComponents
Model Number: 189 with model UnobservedComponents in generation 1 of 10
Template Eval Error: ValueError('Model UnobservedComponents returned NaN for one or more series. fail_on_forecast_nan=True') in model 189: UnobservedComponents
Model Number: 190 with model UnobservedComponents in generation 1 of 10


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 191 with model FBProphet in generation 1 of 10


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 192 with model FBProphet in generation 1 of 10


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 193 with model FBProphet in generation 1 of 10


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 194 with model FBProphet in generation 1 of 10
Model Number: 195 with model GLM in generation 1 of 10
Template Eval Error: TypeError("ufunc 'isfinite' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''") in model 195: GLM
Model Number: 196 with model GLM in generation 1 of 10
Model Number: 197 with model GLM in generation 1 of 10
Template Eval Error: TypeError("ufunc 'isfinite' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''") in model 197: GLM
Model Number: 198 with model GLM in generation 1 of 10


/usr/local/lib/python3.7/dist-packages/statsmodels/genmod/families/family.py:890: RuntimeWarning:

divide by zero encountered in true_divide



Model Number: 199 with model GLS in generation 1 of 10
Model Number: 200 with model GLS in generation 1 of 10
Model Number: 201 with model GLS in generation 1 of 10
Model Number: 202 with model GluonTS in generation 1 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in model 202: GluonTS
Model Number: 203 with model GluonTS in generation 1 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in model 203: GluonTS
Model Number: 204 with model GluonTS in generation 1 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in model 204: GluonTS
Model Number: 205 with model GluonTS in generation 1 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in model 205: GluonTS
Model Number: 206 with model VAR in generat

/usr/local/lib/python3.7/dist-packages/statsmodels/genmod/families/links.py:190: RuntimeWarning:

overflow encountered in exp

/usr/local/lib/python3.7/dist-packages/statsmodels/genmod/families/family.py:889: RuntimeWarning:

divide by zero encountered in true_divide

/usr/local/lib/python3.7/dist-packages/statsmodels/genmod/families/links.py:190: RuntimeWarning:

overflow encountered in exp

/usr/local/lib/python3.7/dist-packages/statsmodels/genmod/families/family.py:889: RuntimeWarning:

divide by zero encountered in true_divide



Model Number: 269 with model GLM in generation 2 of 10
Model Number: 270 with model GLM in generation 2 of 10
Template Eval Error: ValueError('The first guess on the deviance function returned a nan.  This could be a boundary  problem and should be reported.') in model 270: GLM
Model Number: 271 with model FBProphet in generation 2 of 10


/usr/local/lib/python3.7/dist-packages/statsmodels/genmod/families/family.py:1440: RuntimeWarning:

invalid value encountered in log

INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 272 with model FBProphet in generation 2 of 10
Template Eval Error: ValueError("regression_type='User' but no future_regressor passed") in model 272: FBProphet
Model Number: 273 with model FBProphet in generation 2 of 10


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 274 with model FBProphet in generation 2 of 10
Model Number: 275 with model GLS in generation 2 of 10
Model Number: 276 with model GLS in generation 2 of 10
Model Number: 277 with model GLS in generation 2 of 10
Model Number: 278 with model GluonTS in generation 2 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in model 278: GluonTS
Model Number: 279 with model GluonTS in generation 2 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in model 279: GluonTS
Model Number: 280 with model GluonTS in generation 2 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in model 280: GluonTS
Model Number: 281 with model GluonTS in generation 2 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in m

[Parallel(n_jobs=-2)]: Using backend SequentialBackend with 1 concurrent workers.
[Parallel(n_jobs=-2)]: Done 300 out of 300 | elapsed:    0.6s finished
[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.
[Parallel(n_jobs=1)]: Done 300 out of 300 | elapsed:    0.0s finished
/usr/local/lib/python3.7/dist-packages/sklearn/svm/_base.py:1208: ConvergenceWarning:

Liblinear failed to converge, increase the number of iterations.



Model Number: 336 with model DatepartRegression in generation 3 of 10
Template Eval Error: ValueError("regression_type='User' but no future_regressor passed") in model 336: DatepartRegression
Model Number: 337 with model DatepartRegression in generation 3 of 10
Model Number: 338 with model ZeroesNaive in generation 3 of 10
Model Number: 339 with model ZeroesNaive in generation 3 of 10
Model Number: 340 with model ZeroesNaive in generation 3 of 10
Model Number: 341 with model UnobservedComponents in generation 3 of 10
Model Number: 342 with model UnobservedComponents in generation 3 of 10
Model Number: 343 with model UnobservedComponents in generation 3 of 10
Template Eval Error: ValueError("regression_type='User' but no future_regressor supplied") in model 343: UnobservedComponents
Model Number: 344 with model GLM in generation 3 of 10
Template Eval Error: ValueError('The first guess on the deviance function returned a nan.  This could be a boundary  problem and should be reported.') i

/usr/local/lib/python3.7/dist-packages/statsmodels/genmod/families/family.py:1227: RuntimeWarning:

invalid value encountered in log

/usr/local/lib/python3.7/dist-packages/statsmodels/genmod/families/family.py:1227: RuntimeWarning:

invalid value encountered in log

INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Template Eval Error: ValueError('The first guess on the deviance function returned a nan.  This could be a boundary  problem and should be reported.') in model 347: GLM
Model Number: 348 with model FBProphet in generation 3 of 10


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 349 with model FBProphet in generation 3 of 10


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 350 with model FBProphet in generation 3 of 10


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 351 with model FBProphet in generation 3 of 10
Model Number: 352 with model GLS in generation 3 of 10
Model Number: 353 with model GLS in generation 3 of 10
Model Number: 354 with model GLS in generation 3 of 10
Model Number: 355 with model GluonTS in generation 3 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in model 355: GluonTS
Model Number: 356 with model GluonTS in generation 3 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in model 356: GluonTS
Model Number: 357 with model GluonTS in generation 3 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in model 357: GluonTS
Model Number: 358 with model GluonTS in generation 3 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in m

/usr/local/lib/python3.7/dist-packages/sklearn/linear_model/_coordinate_descent.py:648: ConvergenceWarning:

Objective did not converge. You might want to increase the number of iterations, check the scale of the features or consider increasing regularisation. Duality gap: 3.871e+10, tolerance: 2.434e+07



Model Number: 413 with model SeasonalNaive in generation 4 of 10
Model Number: 414 with model SeasonalNaive in generation 4 of 10
Model Number: 415 with model FBProphet in generation 4 of 10


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 416 with model FBProphet in generation 4 of 10


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 417 with model FBProphet in generation 4 of 10


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 418 with model FBProphet in generation 4 of 10
Model Number: 419 with model DatepartRegression in generation 4 of 10
Template Eval Error: ValueError("regression_type='User' but no future_regressor passed") in model 419: DatepartRegression
Model Number: 420 with model DatepartRegression in generation 4 of 10
Template Eval Error: ValueError("regression_type='User' but no future_regressor passed") in model 420: DatepartRegression
Model Number: 421 with model DatepartRegression in generation 4 of 10
Template Eval Error: ValueError('Model DatepartRegression returned NaN for one or more series. fail_on_forecast_nan=True') in model 421: DatepartRegression
Model Number: 422 with model UnobservedComponents in generation 4 of 10


/usr/local/lib/python3.7/dist-packages/sklearn/neighbors/_regression.py:470: UserWarning:

One or more samples have no neighbors within specified radius; predicting NaN.



Model Number: 423 with model UnobservedComponents in generation 4 of 10
Model Number: 424 with model UnobservedComponents in generation 4 of 10
Template Eval Error: ValueError("regression_type='User' but no future_regressor supplied") in model 424: UnobservedComponents
Model Number: 425 with model GLM in generation 4 of 10
Template Eval Error: ValueError('regression_type=user and no future_regressor passed') in model 425: GLM
Model Number: 426 with model GLM in generation 4 of 10
Template Eval Error: ValueError('regression_type=user and no future_regressor passed') in model 426: GLM
Model Number: 427 with model GLM in generation 4 of 10
Template Eval Error: ValueError('regression_type=user and no future_regressor passed') in model 427: GLM
Model Number: 428 with model GLM in generation 4 of 10
Model Number: 429 with model GLS in generation 4 of 10
Model Number: 430 with model GLS in generation 4 of 10
Model Number: 431 with model GLS in generation 4 of 10
Model Number: 432 with model G

INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 488 with model FBProphet in generation 5 of 10


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 489 with model FBProphet in generation 5 of 10


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 490 with model FBProphet in generation 5 of 10
Model Number: 491 with model SeasonalNaive in generation 5 of 10
Model Number: 492 with model SeasonalNaive in generation 5 of 10
Model Number: 493 with model SeasonalNaive in generation 5 of 10
Model Number: 494 with model SeasonalNaive in generation 5 of 10
Model Number: 495 with model GLS in generation 5 of 10
Model Number: 496 with model GLS in generation 5 of 10
Model Number: 497 with model GLS in generation 5 of 10
Model Number: 498 with model DatepartRegression in generation 5 of 10
Model Number: 499 with model DatepartRegression in generation 5 of 10
Template Eval Error: ValueError('Model DatepartRegression returned NaN for one or more series. fail_on_forecast_nan=True') in model 499: DatepartRegression
Model Number: 500 with model DatepartRegression in generation 5 of 10


/usr/local/lib/python3.7/dist-packages/sklearn/neighbors/_regression.py:470: UserWarning:

One or more samples have no neighbors within specified radius; predicting NaN.



Template Eval Error: ValueError('Failed to convert a NumPy array to a Tensor (Unsupported object type int).') in model 500: DatepartRegression
Model Number: 501 with model UnobservedComponents in generation 5 of 10
Model Number: 502 with model UnobservedComponents in generation 5 of 10
Template Eval Error: ValueError('Provided exogenous values are not of the appropriate shape. Required (30, 1), got (30,).') in model 502: UnobservedComponents
Model Number: 503 with model UnobservedComponents in generation 5 of 10
Model Number: 504 with model GLM in generation 5 of 10
Model Number: 505 with model GLM in generation 5 of 10
Template Eval Error: ValueError('The first guess on the deviance function returned a nan.  This could be a boundary  problem and should be reported.') in model 505: GLM
Model Number: 506 with model GLM in generation 5 of 10
Template Eval Error: ValueError('regression_type=user and no future_regressor passed') in model 506: GLM
Model Number: 507 with model GLM in generat

/usr/local/lib/python3.7/dist-packages/statsmodels/genmod/families/family.py:1440: RuntimeWarning:

invalid value encountered in log



Model Number: 510 with model GluonTS in generation 5 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in model 510: GluonTS
Model Number: 511 with model GluonTS in generation 5 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in model 511: GluonTS
Model Number: 512 with model VAR in generation 5 of 10
Template Eval Error: ValueError('Only gave one variable to VAR') in model 512: VAR
Model Number: 513 with model VAR in generation 5 of 10
Template Eval Error: ValueError('Only gave one variable to VAR') in model 513: VAR
Model Number: 514 with model VAR in generation 5 of 10
Template Eval Error: ValueError('Only gave one variable to VAR') in model 514: VAR
Model Number: 515 with model VAR in generation 5 of 10
Template Eval Error: ValueError('Only gave one variable to VAR') in model 515: VAR
Model Number: 516 with model VECM in generation 5 of

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:57: RuntimeWarning:

divide by zero encountered in true_divide

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:58: RuntimeWarning:

divide by zero encountered in true_divide

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:58: RuntimeWarning:

invalid value encountered in true_divide



Model Number: 553 with model MultivariateRegression in generation 6 of 10
Model Number: 554 with model LastValueNaive in generation 6 of 10
Model Number: 555 with model LastValueNaive in generation 6 of 10
Model Number: 556 with model LastValueNaive in generation 6 of 10
Model Number: 557 with model SeasonalNaive in generation 6 of 10
Model Number: 558 with model SeasonalNaive in generation 6 of 10
Model Number: 559 with model SeasonalNaive in generation 6 of 10
Model Number: 560 with model SeasonalNaive in generation 6 of 10
Model Number: 561 with model ETS in generation 6 of 10
ETS error TypeError("__init__() got an unexpected keyword argument 'damped_trend'")
Model Number: 562 with model ETS in generation 6 of 10
ETS error TypeError("__init__() got an unexpected keyword argument 'damped_trend'")
Model Number: 563 with model ETS in generation 6 of 10
ETS error TypeError("__init__() got an unexpected keyword argument 'damped_trend'")


/usr/local/lib/python3.7/dist-packages/sklearn/preprocessing/_data.py:3253: RuntimeWarning:

divide by zero encountered in log

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:57: RuntimeWarning:

invalid value encountered in true_divide

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:58: RuntimeWarning:

divide by zero encountered in true_divide

/usr/local/lib/python3.7/dist-packages/sklearn/preprocessing/_data.py:3196: RuntimeWarning:

divide by zero encountered in power

/usr/local/lib/python3.7/dist-packages/sklearn/preprocessing/_data.py:3196: RuntimeWarning:

divide by zero encountered in power

/usr/local/lib/python3.7/dist-packages/sklearn/preprocessing/_data.py:3196: RuntimeWarning:

divide by zero encountered in power



Model Number: 564 with model ETS in generation 6 of 10
ETS error TypeError("__init__() got an unexpected keyword argument 'damped_trend'")
Model Number: 565 with model AverageValueNaive in generation 6 of 10
Model Number: 566 with model AverageValueNaive in generation 6 of 10
Model Number: 567 with model AverageValueNaive in generation 6 of 10


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.


Model Number: 568 with model FBProphet in generation 6 of 10


INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 569 with model FBProphet in generation 6 of 10


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 570 with model FBProphet in generation 6 of 10
Template Eval Error: ValueError("regression_type='User' but no future_regressor passed") in model 570: FBProphet
Model Number: 571 with model FBProphet in generation 6 of 10
Model Number: 572 with model GLS in generation 6 of 10
Model Number: 573 with model GLS in generation 6 of 10
Model Number: 574 with model GLS in generation 6 of 10
Model Number: 575 with model UnobservedComponents in generation 6 of 10
Model Number: 576 with model UnobservedComponents in generation 6 of 10
Model Number: 577 with model UnobservedComponents in generation 6 of 10
Model Number: 578 with model DatepartRegression in generation 6 of 10
Model Number: 579 with model DatepartRegression in generation 6 of 10


[Parallel(n_jobs=-2)]: Using backend SequentialBackend with 1 concurrent workers.
[Parallel(n_jobs=-2)]: Done 1000 out of 1000 | elapsed:    4.4s finished
[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.
[Parallel(n_jobs=1)]: Done 1000 out of 1000 | elapsed:    0.1s finished


Model Number: 580 with model DatepartRegression in generation 6 of 10


[Parallel(n_jobs=-2)]: Using backend SequentialBackend with 1 concurrent workers.
[Parallel(n_jobs=-2)]: Done 100 out of 100 | elapsed:    0.4s finished
[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.
[Parallel(n_jobs=1)]: Done 100 out of 100 | elapsed:    0.0s finished


Model Number: 581 with model GLM in generation 6 of 10
Template Eval Error: TypeError("ufunc 'isfinite' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''") in model 581: GLM
Model Number: 582 with model GLM in generation 6 of 10
Template Eval Error: TypeError("ufunc 'isfinite' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''") in model 582: GLM
Model Number: 583 with model GLM in generation 6 of 10
Model Number: 584 with model GLM in generation 6 of 10
Model Number: 585 with model GluonTS in generation 6 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in model 585: GluonTS
Model Number: 586 with model GluonTS in generation 6 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible wit

INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 645 with model FBProphet in generation 7 of 10


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 646 with model FBProphet in generation 7 of 10


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 647 with model FBProphet in generation 7 of 10


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 648 with model FBProphet in generation 7 of 10
Model Number: 649 with model GLS in generation 7 of 10
Model Number: 650 with model GLS in generation 7 of 10
Model Number: 651 with model GLS in generation 7 of 10
Model Number: 652 with model UnobservedComponents in generation 7 of 10
Template Eval Error: ValueError('Provided exogenous values are not of the appropriate shape. Required (30, 1), got (30,).') in model 652: UnobservedComponents
Model Number: 653 with model UnobservedComponents in generation 7 of 10
Template Eval Error: ValueError('Provided exogenous values are not of the appropriate shape. Required (30, 1), got (30,).') in model 653: UnobservedComponents
Model Number: 654 with model UnobservedComponents in generation 7 of 10
Template Eval Error: ValueError("regression_type='User' but no future_regressor supplied") in model 654: UnobservedComponents
Model Number: 655 with model DatepartRegression in generation 7 of 10
Model Number: 656 with model DatepartRegress

/usr/local/lib/python3.7/dist-packages/sklearn/neighbors/_regression.py:470: UserWarning:

One or more samples have no neighbors within specified radius; predicting NaN.

/usr/local/lib/python3.7/dist-packages/sklearn/neighbors/_regression.py:470: UserWarning:

One or more samples have no neighbors within specified radius; predicting NaN.



Template Eval Error: ValueError('Model DatepartRegression returned NaN for one or more series. fail_on_forecast_nan=True') in model 657: DatepartRegression
Model Number: 658 with model GLM in generation 7 of 10
Template Eval Error: ValueError('The first guess on the deviance function returned a nan.  This could be a boundary  problem and should be reported.') in model 658: GLM
Model Number: 659 with model GLM in generation 7 of 10
Model Number: 660 with model GLM in generation 7 of 10


/usr/local/lib/python3.7/dist-packages/statsmodels/genmod/families/family.py:1440: RuntimeWarning:

invalid value encountered in log

/usr/local/lib/python3.7/dist-packages/statsmodels/genmod/families/family.py:890: RuntimeWarning:

divide by zero encountered in true_divide



Model Number: 661 with model GLM in generation 7 of 10
Template Eval Error: ValueError('The first guess on the deviance function returned a nan.  This could be a boundary  problem and should be reported.') in model 661: GLM
Model Number: 662 with model GluonTS in generation 7 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in model 662: GluonTS
Model Number: 663 with model GluonTS in generation 7 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in model 663: GluonTS
Model Number: 664 with model GluonTS in generation 7 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in model 664: GluonTS
Model Number: 665 with model GluonTS in generation 7 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in mod

/usr/local/lib/python3.7/dist-packages/statsmodels/genmod/families/family.py:1227: RuntimeWarning:

invalid value encountered in log



Template Eval Error: ModuleNotFoundError("No module named 'statsmodels.tsa.forecasting'") in model 674: Theta
Model Number: 675 with model Theta in generation 7 of 10
Template Eval Error: ModuleNotFoundError("No module named 'statsmodels.tsa.forecasting'") in model 675: Theta
Model Number: 676 with model Theta in generation 7 of 10
Template Eval Error: ModuleNotFoundError("No module named 'statsmodels.tsa.forecasting'") in model 676: Theta
Model Number: 677 with model Theta in generation 7 of 10
Template Eval Error: ModuleNotFoundError("No module named 'statsmodels.tsa.forecasting'") in model 677: Theta
Model Number: 678 with model ARDL in generation 7 of 10
Template Eval Error: Exception('Transformer STLFilter failed on fit') in model 678: ARDL
Model Number: 679 with model ARDL in generation 7 of 10
Template Eval Error: ImportError("cannot import name 'ARDL' from 'statsmodels.tsa.api' (/usr/local/lib/python3.7/dist-packages/statsmodels/tsa/api.py)") in model 679: ARDL
Model Number: 68

/usr/local/lib/python3.7/dist-packages/autots/models/base.py:344: RuntimeWarning:

overflow encountered in square



Model Number: 705 with model MultivariateRegression in generation 8 of 10
Model Number: 706 with model MultivariateRegression in generation 8 of 10
Model Number: 707 with model LastValueNaive in generation 8 of 10
Model Number: 708 with model LastValueNaive in generation 8 of 10
Model Number: 709 with model LastValueNaive in generation 8 of 10
Model Number: 710 with model SeasonalNaive in generation 8 of 10
Model Number: 711 with model SeasonalNaive in generation 8 of 10
Model Number: 712 with model SeasonalNaive in generation 8 of 10
Model Number: 713 with model SeasonalNaive in generation 8 of 10
Model Number: 714 with model ETS in generation 8 of 10
ETS error TypeError("__init__() got an unexpected keyword argument 'damped_trend'")
Model Number: 715 with model ETS in generation 8 of 10
ETS error TypeError("__init__() got an unexpected keyword argument 'damped_trend'")
ETS failed on Close with ValueError('endog must be strictly positive when using multiplicative trend or seasonal com

INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 718 with model FBProphet in generation 8 of 10
Template Eval Error: ValueError("regression_type='User' but no future_regressor passed") in model 718: FBProphet
Model Number: 719 with model FBProphet in generation 8 of 10
Template Eval Error: ValueError("regression_type='User' but no future_regressor passed") in model 719: FBProphet
Model Number: 720 with model FBProphet in generation 8 of 10


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 721 with model FBProphet in generation 8 of 10
Model Number: 722 with model AverageValueNaive in generation 8 of 10
Model Number: 723 with model AverageValueNaive in generation 8 of 10
Model Number: 724 with model AverageValueNaive in generation 8 of 10
Model Number: 725 with model GLS in generation 8 of 10
Model Number: 726 with model GLS in generation 8 of 10
Model Number: 727 with model GLS in generation 8 of 10
Model Number: 728 with model UnobservedComponents in generation 8 of 10
Model Number: 729 with model UnobservedComponents in generation 8 of 10
Template Eval Error: ValueError('Provided exogenous values are not of the appropriate shape. Required (30, 1), got (30,).') in model 729: UnobservedComponents
Model Number: 730 with model UnobservedComponents in generation 8 of 10
Model Number: 731 with model DatepartRegression in generation 8 of 10


[Parallel(n_jobs=-2)]: Using backend SequentialBackend with 1 concurrent workers.
[Parallel(n_jobs=-2)]: Done 100 out of 100 | elapsed:    0.1s finished
[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.
[Parallel(n_jobs=1)]: Done 100 out of 100 | elapsed:    0.0s finished


Model Number: 732 with model DatepartRegression in generation 8 of 10
Template Eval Error: ValueError("regression_type='User' but no future_regressor passed") in model 732: DatepartRegression
Model Number: 733 with model DatepartRegression in generation 8 of 10


[Parallel(n_jobs=-2)]: Using backend SequentialBackend with 1 concurrent workers.
[Parallel(n_jobs=-2)]: Done 300 out of 300 | elapsed:    1.1s finished
[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.
[Parallel(n_jobs=1)]: Done 300 out of 300 | elapsed:    0.0s finished


Model Number: 734 with model GLM in generation 8 of 10
Model Number: 735 with model GLM in generation 8 of 10
Template Eval Error: ValueError('regression_type=user and no future_regressor passed') in model 735: GLM
Model Number: 736 with model GLM in generation 8 of 10
Template Eval Error: ValueError('The first guess on the deviance function returned a nan.  This could be a boundary  problem and should be reported.') in model 736: GLM
Model Number: 737 with model GluonTS in generation 8 of 10
Template Eval Error: Exception('Transformer STLFilter failed on fit') in model 737: GluonTS
Model Number: 738 with model GluonTS in generation 8 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in model 738: GluonTS
Model Number: 739 with model GluonTS in generation 8 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in model 739: GluonTS
Model Number: 

/usr/local/lib/python3.7/dist-packages/statsmodels/genmod/families/family.py:1227: RuntimeWarning:

invalid value encountered in log



Template Eval Error: ValueError('Only gave one variable to VECM') in model 747: VECM
Model Number: 748 with model VECM in generation 8 of 10
Template Eval Error: ValueError('Only gave one variable to VECM') in model 748: VECM
Model Number: 749 with model Theta in generation 8 of 10
Template Eval Error: ModuleNotFoundError("No module named 'statsmodels.tsa.forecasting'") in model 749: Theta
Model Number: 750 with model Theta in generation 8 of 10
Template Eval Error: ModuleNotFoundError("No module named 'statsmodels.tsa.forecasting'") in model 750: Theta
Model Number: 751 with model Theta in generation 8 of 10
Template Eval Error: ModuleNotFoundError("No module named 'statsmodels.tsa.forecasting'") in model 751: Theta
Model Number: 752 with model ARDL in generation 8 of 10
Template Eval Error: ImportError("cannot import name 'ARDL' from 'statsmodels.tsa.api' (/usr/local/lib/python3.7/dist-packages/statsmodels/tsa/api.py)") in model 752: ARDL
Model Number: 753 with model ARDL in generati

/usr/local/lib/python3.7/dist-packages/sklearn/linear_model/_coordinate_descent.py:648: ConvergenceWarning:

Objective did not converge. You might want to increase the number of iterations, check the scale of the features or consider increasing regularisation. Duality gap: 2.691e+01, tolerance: 1.768e-02



Model Number: 773 with model ZeroesNaive in generation 9 of 10
Model Number: 774 with model ZeroesNaive in generation 9 of 10
Model Number: 775 with model WindowRegression in generation 9 of 10
Template Eval Error: LightGBMError('[tweedie]: at least one target label is negative') in model 775: WindowRegression
Model Number: 776 with model WindowRegression in generation 9 of 10
Model Number: 777 with model WindowRegression in generation 9 of 10
Template Eval Error: ValueError("regression_type='User' but no future_regressor passed") in model 777: WindowRegression
Model Number: 778 with model MultivariateRegression in generation 9 of 10


/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:57: RuntimeWarning:

divide by zero encountered in true_divide

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:58: RuntimeWarning:

divide by zero encountered in true_divide

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:58: RuntimeWarning:

invalid value encountered in true_divide

/usr/local/lib/python3.7/dist-packages/sklearn/neighbors/_regression.py:470: UserWarning:

One or more samples have no neighbors within specified radius; predicting NaN.

/usr/local/lib/python3.7/dist-packages/sklearn/neighbors/_regression.py:470: UserWarning:

One or more samples have no neighbors within specified radius; predicting NaN.



Model Number: 779 with model MultivariateRegression in generation 9 of 10


/usr/local/lib/python3.7/dist-packages/sklearn/neighbors/_regression.py:470: UserWarning:

One or more samples have no neighbors within specified radius; predicting NaN.

/usr/local/lib/python3.7/dist-packages/sklearn/neighbors/_regression.py:470: UserWarning:

One or more samples have no neighbors within specified radius; predicting NaN.

/usr/local/lib/python3.7/dist-packages/sklearn/neighbors/_regression.py:470: UserWarning:

One or more samples have no neighbors within specified radius; predicting NaN.

/usr/local/lib/python3.7/dist-packages/sklearn/neighbors/_regression.py:470: UserWarning:

One or more samples have no neighbors within specified radius; predicting NaN.

/usr/local/lib/python3.7/dist-packages/sklearn/neighbors/_regression.py:470: UserWarning:

One or more samples have no neighbors within specified radius; predicting NaN.

/usr/local/lib/python3.7/dist-packages/sklearn/neighbors/_regression.py:470: UserWarning:

One or more samples have no neighbors within specified

Template Eval Error: ValueError('Model MultivariateRegression returned NaN for one or more series. fail_on_forecast_nan=True') in model 779: MultivariateRegression
Model Number: 780 with model MultivariateRegression in generation 9 of 10
Template Eval Error: ValueError("regression_type='User' but not future_regressor supplied.") in model 780: MultivariateRegression
Model Number: 781 with model MultivariateRegression in generation 9 of 10
Model Number: 782 with model LastValueNaive in generation 9 of 10
Model Number: 783 with model LastValueNaive in generation 9 of 10
Model Number: 784 with model LastValueNaive in generation 9 of 10
Model Number: 785 with model SeasonalNaive in generation 9 of 10
Model Number: 786 with model SeasonalNaive in generation 9 of 10
Model Number: 787 with model SeasonalNaive in generation 9 of 10
Model Number: 788 with model SeasonalNaive in generation 9 of 10
Model Number: 789 with model ETS in generation 9 of 10
ETS error TypeError("__init__() got an unexpe

INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 793 with model FBProphet in generation 9 of 10


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 794 with model FBProphet in generation 9 of 10


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 795 with model FBProphet in generation 9 of 10
Model Number: 796 with model FBProphet in generation 9 of 10
Template Eval Error: ValueError("regression_type='User' but no future_regressor passed") in model 796: FBProphet
Model Number: 797 with model AverageValueNaive in generation 9 of 10
Model Number: 798 with model AverageValueNaive in generation 9 of 10
Model Number: 799 with model AverageValueNaive in generation 9 of 10
Model Number: 800 with model GLS in generation 9 of 10
Model Number: 801 with model GLS in generation 9 of 10
Model Number: 802 with model UnobservedComponents in generation 9 of 10
Template Eval Error: ValueError("regression_type='User' but no future_regressor supplied") in model 802: UnobservedComponents
Model Number: 803 with model UnobservedComponents in generation 9 of 10
Model Number: 804 with model UnobservedComponents in generation 9 of 10
Model Number: 805 with model DatepartRegression in generation 9 of 10
Template Eval Error: ValueError('Mod

/usr/local/lib/python3.7/dist-packages/sklearn/neighbors/_regression.py:470: UserWarning:

One or more samples have no neighbors within specified radius; predicting NaN.



Model Number: 807 with model DatepartRegression in generation 9 of 10
Template Eval Error: ValueError("regression_type='User' but no future_regressor passed") in model 807: DatepartRegression
Model Number: 808 with model GLM in generation 9 of 10
Model Number: 809 with model GLM in generation 9 of 10
Template Eval Error: ValueError('NaN, inf or invalid value detected in weights, estimation infeasible.') in model 809: GLM
Model Number: 810 with model GLM in generation 9 of 10
Model Number: 811 with model GLM in generation 9 of 10
Template Eval Error: ValueError('The first guess on the deviance function returned a nan.  This could be a boundary  problem and should be reported.') in model 811: GLM
Model Number: 812 with model GluonTS in generation 9 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in model 812: GluonTS
Model Number: 813 with model GluonTS in generation 9 of 10
Template Eval Error: ImportError('Gluon

/usr/local/lib/python3.7/dist-packages/statsmodels/genmod/families/family.py:428: RuntimeWarning:

divide by zero encountered in true_divide

/usr/local/lib/python3.7/dist-packages/statsmodels/genmod/families/family.py:134: RuntimeWarning:

divide by zero encountered in true_divide

/usr/local/lib/python3.7/dist-packages/statsmodels/genmod/families/family.py:1227: RuntimeWarning:

divide by zero encountered in log

/usr/local/lib/python3.7/dist-packages/statsmodels/genmod/families/family.py:1227: RuntimeWarning:

invalid value encountered in multiply



Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in model 814: GluonTS
Model Number: 815 with model GluonTS in generation 9 of 10
Template Eval Error: ImportError('GluonTS installation not found or installed version is incompatible with AutoTS.') in model 815: GluonTS
Model Number: 816 with model VAR in generation 9 of 10
Template Eval Error: ValueError('Only gave one variable to VAR') in model 816: VAR
Model Number: 817 with model VAR in generation 9 of 10
Template Eval Error: ValueError('Only gave one variable to VAR') in model 817: VAR
Model Number: 818 with model VAR in generation 9 of 10
Template Eval Error: ValueError('Only gave one variable to VAR') in model 818: VAR
Model Number: 819 with model VAR in generation 9 of 10
Template Eval Error: ValueError('Only gave one variable to VAR') in model 819: VAR
Model Number: 820 with model VECM in generation 9 of 10
Template Eval Error: ValueError('Only gave one variable

[Parallel(n_jobs=-2)]: Using backend SequentialBackend with 1 concurrent workers.


Model Number: 853 with model WindowRegression in generation 10 of 10
Template Eval Error: ValueError("regression_type='User' but no future_regressor passed") in model 853: WindowRegression
Model Number: 854 with model LastValueNaive in generation 10 of 10
Model Number: 855 with model LastValueNaive in generation 10 of 10


[Parallel(n_jobs=-2)]: Done 300 out of 300 | elapsed:    1.6s finished
[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.
[Parallel(n_jobs=1)]: Done 300 out of 300 | elapsed:    0.0s finished


Model Number: 856 with model LastValueNaive in generation 10 of 10
Model Number: 857 with model MultivariateRegression in generation 10 of 10
Model Number: 858 with model MultivariateRegression in generation 10 of 10
Template Eval Error: ValueError("regression_type='User' but not future_regressor supplied.") in model 858: MultivariateRegression
Model Number: 859 with model MultivariateRegression in generation 10 of 10
Model Number: 860 with model MultivariateRegression in generation 10 of 10
Template Eval Error: ValueError("regression_type='User' but not future_regressor supplied.") in model 860: MultivariateRegression
Model Number: 861 with model SeasonalNaive in generation 10 of 10
Model Number: 862 with model SeasonalNaive in generation 10 of 10
Model Number: 863 with model SeasonalNaive in generation 10 of 10
Model Number: 864 with model SeasonalNaive in generation 10 of 10
Model Number: 865 with model ETS in generation 10 of 10
ETS error TypeError("__init__() got an unexpected key

INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 872 with model FBProphet in generation 10 of 10
Template Eval Error: ValueError("regression_type='User' but no future_regressor passed") in model 872: FBProphet
Model Number: 873 with model AverageValueNaive in generation 10 of 10
Model Number: 874 with model AverageValueNaive in generation 10 of 10
Model Number: 875 with model AverageValueNaive in generation 10 of 10
Model Number: 876 with model GLS in generation 10 of 10
Model Number: 877 with model GLS in generation 10 of 10
Model Number: 878 with model GLS in generation 10 of 10
Model Number: 879 with model Ensemble in generation 11 of 0
Model Number: 880 with model Ensemble in generation 11 of 0
Model Number: 881 with model Ensemble in generation 11 of 0
ETS error TypeError("__init__() got an unexpected keyword argument 'damped_trend'")
Model Number: 882 with model Ensemble in generation 11 of 0


/usr/local/lib/python3.7/dist-packages/sklearn/preprocessing/_data.py:3253: RuntimeWarning:

divide by zero encountered in log

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:57: RuntimeWarning:

invalid value encountered in true_divide

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:58: RuntimeWarning:

divide by zero encountered in true_divide

/usr/local/lib/python3.7/dist-packages/sklearn/preprocessing/_data.py:3196: RuntimeWarning:

divide by zero encountered in power

/usr/local/lib/python3.7/dist-packages/sklearn/preprocessing/_data.py:3196: RuntimeWarning:

divide by zero encountered in power

/usr/local/lib/python3.7/dist-packages/sklearn/preprocessing/_data.py:3196: RuntimeWarning:

divide by zero encountered in power

/usr/local/lib/python3.7/dist-packages/sklearn/preprocessing/_data.py:3253: RuntimeWarning:

divide by zero encountered in log



ETS error TypeError("__init__() got an unexpected keyword argument 'damped_trend'")


/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:57: RuntimeWarning:

invalid value encountered in true_divide

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:58: RuntimeWarning:

divide by zero encountered in true_divide

/usr/local/lib/python3.7/dist-packages/sklearn/preprocessing/_data.py:3196: RuntimeWarning:

divide by zero encountered in power

/usr/local/lib/python3.7/dist-packages/sklearn/preprocessing/_data.py:3196: RuntimeWarning:

divide by zero encountered in power

/usr/local/lib/python3.7/dist-packages/sklearn/preprocessing/_data.py:3196: RuntimeWarning:

divide by zero encountered in power



Model Number: 883 with model Ensemble in generation 11 of 0
Model Number: 884 with model Ensemble in generation 11 of 0
Model Number: 885 with model Ensemble in generation 11 of 0
Model Number: 886 with model Ensemble in generation 11 of 0
Validation Round: 1
Model Number: 1 of 133 with model Ensemble for Validation 1
1 - Ensemble with avg smape 6.31: 
Model Number: 2 of 133 with model Ensemble for Validation 1
2 - Ensemble with avg smape 6.09: 
Model Number: 3 of 133 with model Ensemble for Validation 1
3 - Ensemble with avg smape 6.12: 
Model Number: 4 of 133 with model Ensemble for Validation 1
4 - Ensemble with avg smape 6.12: 
Model Number: 5 of 133 with model Ensemble for Validation 1
5 - Ensemble with avg smape 6.6: 
Model Number: 6 of 133 with model NVAR for Validation 1
6 - NVAR with avg smape 6.55: 
Model Number: 7 of 133 with model Ensemble for Validation 1
7 - Ensemble with avg smape 6.58: 
Model Number: 8 of 133 with model NVAR for Validation 1
8 - NVAR with avg smape 6.6:

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:57: RuntimeWarning:

divide by zero encountered in true_divide

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:58: RuntimeWarning:

divide by zero encountered in true_divide

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:58: RuntimeWarning:

invalid value encountered in true_divide

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:57: RuntimeWarning:

invalid value encountered in add



45 - MultivariateRegression with avg smape 47.68: 
Model Number: 46 of 133 with model MultivariateRegression for Validation 1
46 - MultivariateRegression with avg smape 9.51: 
Model Number: 47 of 133 with model MultivariateRegression for Validation 1
47 - MultivariateRegression with avg smape 9.39: 
Model Number: 48 of 133 with model MultivariateRegression for Validation 1
48 - MultivariateRegression with avg smape 19.77: 
Model Number: 49 of 133 with model SectionalMotif for Validation 1
49 - SectionalMotif with avg smape 13.0: 
Model Number: 50 of 133 with model WindowRegression for Validation 1
50 - WindowRegression with avg smape 7.3: 
Model Number: 51 of 133 with model WindowRegression for Validation 1
51 - WindowRegression with avg smape 7.28: 
Model Number: 52 of 133 with model MultivariateRegression for Validation 1
52 - MultivariateRegression with avg smape 19.2: 
Model Number: 53 of 133 with model MultivariateRegression for Validation 1
53 - MultivariateRegression with avg sm

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:57: RuntimeWarning:

divide by zero encountered in true_divide

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:58: RuntimeWarning:

divide by zero encountered in true_divide

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:58: RuntimeWarning:

invalid value encountered in true_divide



56 - WindowRegression with avg smape 6.7: 
Model Number: 57 of 133 with model LastValueNaive for Validation 1
57 - LastValueNaive with avg smape 8.64: 
Model Number: 58 of 133 with model LastValueNaive for Validation 1
58 - LastValueNaive with avg smape 8.64: 
Model Number: 59 of 133 with model LastValueNaive for Validation 1
59 - LastValueNaive with avg smape 8.27: 
Model Number: 60 of 133 with model ETS for Validation 1
ETS error TypeError("__init__() got an unexpected keyword argument 'damped_trend'")
60 - ETS with avg smape 8.63: 
Model Number: 61 of 133 with model FBProphet for Validation 1


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


61 - FBProphet with avg smape 25.89: 
Model Number: 62 of 133 with model SeasonalNaive for Validation 1
62 - SeasonalNaive with avg smape 7.19: 
Model Number: 63 of 133 with model AverageValueNaive for Validation 1
63 - AverageValueNaive with avg smape 15.45: 
Model Number: 64 of 133 with model AverageValueNaive for Validation 1
64 - AverageValueNaive with avg smape 15.45: 
Model Number: 65 of 133 with model AverageValueNaive for Validation 1
65 - AverageValueNaive with avg smape 15.45: 
Model Number: 66 of 133 with model MultivariateMotif for Validation 1
66 - MultivariateMotif with avg smape 36.66: 
Model Number: 67 of 133 with model WindowRegression for Validation 1
67 - WindowRegression with avg smape 7.52: 
Model Number: 68 of 133 with model LastValueNaive for Validation 1
68 - LastValueNaive with avg smape 12.05: 
Model Number: 69 of 133 with model MultivariateMotif for Validation 1
69 - MultivariateMotif with avg smape 9.98: 
Model Number: 70 of 133 with model LastValueNaive for

INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


Model Number: 80 of 133 with model FBProphet for Validation 1


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


80 - FBProphet with avg smape 26.73: 
Model Number: 81 of 133 with model FBProphet for Validation 1
81 - FBProphet with avg smape 26.12: 
Model Number: 82 of 133 with model SeasonalNaive for Validation 1
82 - SeasonalNaive with avg smape 7.29: 
Model Number: 83 of 133 with model ZeroesNaive for Validation 1
83 - ZeroesNaive with avg smape 8.32: 
Model Number: 84 of 133 with model LastValueNaive for Validation 1
84 - LastValueNaive with avg smape 8.21: 
Model Number: 85 of 133 with model FBProphet for Validation 1


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


85 - FBProphet with avg smape 25.64: 
Model Number: 86 of 133 with model SeasonalNaive for Validation 1
86 - SeasonalNaive with avg smape 10.94: 
Model Number: 87 of 133 with model SeasonalNaive for Validation 1
87 - SeasonalNaive with avg smape 10.94: 
Model Number: 88 of 133 with model LastValueNaive for Validation 1
88 - LastValueNaive with avg smape 12.03: 
Model Number: 89 of 133 with model AverageValueNaive for Validation 1
89 - AverageValueNaive with avg smape 6.33: 
Model Number: 90 of 133 with model AverageValueNaive for Validation 1
90 - AverageValueNaive with avg smape 6.11: 
Model Number: 91 of 133 with model SeasonalNaive for Validation 1
91 - SeasonalNaive with avg smape 10.21: 
Model Number: 92 of 133 with model AverageValueNaive for Validation 1
92 - AverageValueNaive with avg smape 6.06: 
Model Number: 93 of 133 with model AverageValueNaive for Validation 1


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


93 - AverageValueNaive with avg smape 6.07: 
Model Number: 94 of 133 with model FBProphet for Validation 1


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


94 - FBProphet with avg smape 25.2: 
Model Number: 95 of 133 with model FBProphet for Validation 1


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


95 - FBProphet with avg smape 24.98: 
Model Number: 96 of 133 with model FBProphet for Validation 1
96 - FBProphet with avg smape 20.76: 
Model Number: 97 of 133 with model SeasonalNaive for Validation 1
97 - SeasonalNaive with avg smape 6.29: 
Model Number: 98 of 133 with model GLS for Validation 1
98 - GLS with avg smape 4.71: 
Model Number: 99 of 133 with model GLS for Validation 1
99 - GLS with avg smape 4.71: 
Model Number: 100 of 133 with model FBProphet for Validation 1


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


100 - FBProphet with avg smape 23.91: 
Model Number: 101 of 133 with model SeasonalNaive for Validation 1
101 - SeasonalNaive with avg smape 6.88: 
Model Number: 102 of 133 with model GLS for Validation 1
102 - GLS with avg smape 4.71: 
Model Number: 103 of 133 with model SeasonalNaive for Validation 1
103 - SeasonalNaive with avg smape 6.49: 
Model Number: 104 of 133 with model GLS for Validation 1
104 - GLS with avg smape 7.87: 
Model Number: 105 of 133 with model UnobservedComponents for Validation 1
105 - UnobservedComponents with avg smape 8.23: 
Model Number: 106 of 133 with model UnobservedComponents for Validation 1
106 - UnobservedComponents with avg smape 8.3: 
Model Number: 107 of 133 with model DatepartRegression for Validation 1
107 - DatepartRegression with avg smape 4.03: 
Model Number: 108 of 133 with model GLM for Validation 1
108 - GLM with avg smape 5.67: 
Model Number: 109 of 133 with model GLM for Validation 1
109 - GLM with avg smape 5.67: 
Model Number: 110 of 13

/usr/local/lib/python3.7/dist-packages/sklearn/svm/_base.py:1208: ConvergenceWarning:

Liblinear failed to converge, increase the number of iterations.

[Parallel(n_jobs=-2)]: Using backend SequentialBackend with 1 concurrent workers.
[Parallel(n_jobs=-2)]: Done 1000 out of 1000 | elapsed:    4.6s finished
[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.
[Parallel(n_jobs=1)]: Done 1000 out of 1000 | elapsed:    0.1s finished


119 - DatepartRegression with avg smape 9.34: 
Model Number: 120 of 133 with model GLS for Validation 1
120 - GLS with avg smape 15.86: 
Model Number: 121 of 133 with model GLM for Validation 1
121 - GLM with avg smape 7.17: 
Model Number: 122 of 133 with model GLM for Validation 1
122 - GLM with avg smape 7.15: 
Model Number: 123 of 133 with model UnobservedComponents for Validation 1
123 - UnobservedComponents with avg smape 7.04: 
Model Number: 124 of 133 with model UnobservedComponents for Validation 1
124 - UnobservedComponents with avg smape 7.36: 
Model Number: 125 of 133 with model GLS for Validation 1
125 - GLS with avg smape 12.11: 
Model Number: 126 of 133 with model DatepartRegression for Validation 1


[Parallel(n_jobs=-2)]: Using backend SequentialBackend with 1 concurrent workers.
[Parallel(n_jobs=-2)]: Done 300 out of 300 | elapsed:    0.6s finished
[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.
[Parallel(n_jobs=1)]: Done 300 out of 300 | elapsed:    0.0s finished


126 - DatepartRegression with avg smape 6.95: 
Model Number: 127 of 133 with model DatepartRegression for Validation 1


[Parallel(n_jobs=-2)]: Using backend SequentialBackend with 1 concurrent workers.
[Parallel(n_jobs=-2)]: Done 100 out of 100 | elapsed:    0.4s finished
[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.
[Parallel(n_jobs=1)]: Done 100 out of 100 | elapsed:    0.0s finished


127 - DatepartRegression with avg smape 4.76: 
Model Number: 128 of 133 with model GLS for Validation 1
128 - GLS with avg smape 16.89: 
Model Number: 129 of 133 with model DatepartRegression for Validation 1
129 - DatepartRegression with avg smape 10.29: 
Model Number: 130 of 133 with model DatepartRegression for Validation 1
130 - DatepartRegression with avg smape 6.64: 
Model Number: 131 of 133 with model DatepartRegression for Validation 1
131 - DatepartRegression with avg smape 16.42: 
Model Number: 132 of 133 with model GLM for Validation 1
132 - GLM with avg smape 12.67: 
Model Number: 133 of 133 with model GLM for Validation 1
133 - GLM with avg smape 12.9: 
Validation Round: 2
Model Number: 1 of 133 with model Ensemble for Validation 2


/usr/local/lib/python3.7/dist-packages/statsmodels/genmod/families/links.py:190: RuntimeWarning:

overflow encountered in exp

/usr/local/lib/python3.7/dist-packages/statsmodels/genmod/families/family.py:889: RuntimeWarning:

divide by zero encountered in true_divide



1 - Ensemble with avg smape 18.03: 
Model Number: 2 of 133 with model Ensemble for Validation 2
2 - Ensemble with avg smape 16.16: 
Model Number: 3 of 133 with model Ensemble for Validation 2
3 - Ensemble with avg smape 16.32: 
Model Number: 4 of 133 with model Ensemble for Validation 2
4 - Ensemble with avg smape 16.32: 
Model Number: 5 of 133 with model Ensemble for Validation 2
5 - Ensemble with avg smape 15.47: 
Model Number: 6 of 133 with model NVAR for Validation 2
6 - NVAR with avg smape 15.46: 
Model Number: 7 of 133 with model Ensemble for Validation 2
7 - Ensemble with avg smape 15.65: 
Model Number: 8 of 133 with model NVAR for Validation 2
8 - NVAR with avg smape 15.86: 
Model Number: 9 of 133 with model NVAR for Validation 2
9 - NVAR with avg smape 15.64: 
Model Number: 10 of 133 with model NVAR for Validation 2
10 - NVAR with avg smape 14.76: 
Model Number: 11 of 133 with model NVAR for Validation 2
11 - NVAR with avg smape 14.77: 
Model Number: 12 of 133 with model NVAR 

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:57: RuntimeWarning:

divide by zero encountered in true_divide

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:58: RuntimeWarning:

divide by zero encountered in true_divide

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:58: RuntimeWarning:

invalid value encountered in true_divide



45 - MultivariateRegression with avg smape 7.38: 
Model Number: 46 of 133 with model MultivariateRegression for Validation 2
46 - MultivariateRegression with avg smape 15.59: 
Model Number: 47 of 133 with model MultivariateRegression for Validation 2
47 - MultivariateRegression with avg smape 14.77: 
Model Number: 48 of 133 with model MultivariateRegression for Validation 2
48 - MultivariateRegression with avg smape 13.41: 
Model Number: 49 of 133 with model SectionalMotif for Validation 2
49 - SectionalMotif with avg smape 13.13: 
Model Number: 50 of 133 with model WindowRegression for Validation 2
50 - WindowRegression with avg smape 14.06: 
Model Number: 51 of 133 with model WindowRegression for Validation 2
51 - WindowRegression with avg smape 22.36: 
Model Number: 52 of 133 with model MultivariateRegression for Validation 2
52 - MultivariateRegression with avg smape 12.72: 
Model Number: 53 of 133 with model MultivariateRegression for Validation 2
53 - MultivariateRegression with 

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:57: RuntimeWarning:

divide by zero encountered in true_divide

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:58: RuntimeWarning:

divide by zero encountered in true_divide

/usr/local/lib/python3.7/dist-packages/autots/tools/probabilistic.py:58: RuntimeWarning:

invalid value encountered in true_divide



55 - MultivariateRegression with avg smape 7.51: 
Model Number: 56 of 133 with model WindowRegression for Validation 2
56 - WindowRegression with avg smape 19.4: 
Model Number: 57 of 133 with model LastValueNaive for Validation 2
57 - LastValueNaive with avg smape 18.23: 
Model Number: 58 of 133 with model LastValueNaive for Validation 2
58 - LastValueNaive with avg smape 18.23: 
Model Number: 59 of 133 with model LastValueNaive for Validation 2
59 - LastValueNaive with avg smape 18.4: 
Model Number: 60 of 133 with model ETS for Validation 2
ETS error TypeError("__init__() got an unexpected keyword argument 'damped_trend'")
60 - ETS with avg smape 19.39: 
Model Number: 61 of 133 with model FBProphet for Validation 2


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


61 - FBProphet with avg smape 36.67: 
Model Number: 62 of 133 with model SeasonalNaive for Validation 2
62 - SeasonalNaive with avg smape 14.79: 
Model Number: 63 of 133 with model AverageValueNaive for Validation 2
63 - AverageValueNaive with avg smape 25.62: 
Model Number: 64 of 133 with model AverageValueNaive for Validation 2
64 - AverageValueNaive with avg smape 25.62: 
Model Number: 65 of 133 with model AverageValueNaive for Validation 2
65 - AverageValueNaive with avg smape 25.62: 
Model Number: 66 of 133 with model MultivariateMotif for Validation 2
66 - MultivariateMotif with avg smape 21.16: 
Model Number: 67 of 133 with model WindowRegression for Validation 2
67 - WindowRegression with avg smape 13.96: 
Model Number: 68 of 133 with model LastValueNaive for Validation 2
68 - LastValueNaive with avg smape 14.11: 
Model Number: 69 of 133 with model MultivariateMotif for Validation 2
69 - MultivariateMotif with avg smape 10.53: 
Model Number: 70 of 133 with model LastValueNaive 

INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.



Model Number: 80 of 133 with model FBProphet for Validation 2


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


80 - FBProphet with avg smape 39.09: 
Model Number: 81 of 133 with model FBProphet for Validation 2
81 - FBProphet with avg smape 38.93: 
Model Number: 82 of 133 with model SeasonalNaive for Validation 2
82 - SeasonalNaive with avg smape 17.09: 
Model Number: 83 of 133 with model ZeroesNaive for Validation 2
83 - ZeroesNaive with avg smape 13.36: 
Model Number: 84 of 133 with model LastValueNaive for Validation 2
84 - LastValueNaive with avg smape 18.56: 
Model Number: 85 of 133 with model FBProphet for Validation 2


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


85 - FBProphet with avg smape 39.2: 
Model Number: 86 of 133 with model SeasonalNaive for Validation 2
86 - SeasonalNaive with avg smape 31.64: 
Model Number: 87 of 133 with model SeasonalNaive for Validation 2
87 - SeasonalNaive with avg smape 31.64: 
Model Number: 88 of 133 with model LastValueNaive for Validation 2
88 - LastValueNaive with avg smape 13.91: 
Model Number: 89 of 133 with model AverageValueNaive for Validation 2
89 - AverageValueNaive with avg smape 6.8: 
Model Number: 90 of 133 with model AverageValueNaive for Validation 2
90 - AverageValueNaive with avg smape 6.91: 
Model Number: 91 of 133 with model SeasonalNaive for Validation 2
91 - SeasonalNaive with avg smape 20.97: 
Model Number: 92 of 133 with model AverageValueNaive for Validation 2
92 - AverageValueNaive with avg smape 6.89: 
Model Number: 93 of 133 with model AverageValueNaive for Validation 2


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


93 - AverageValueNaive with avg smape 6.86: 
Model Number: 94 of 133 with model FBProphet for Validation 2


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


94 - FBProphet with avg smape 38.79: 
Model Number: 95 of 133 with model FBProphet for Validation 2


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


95 - FBProphet with avg smape 39.22: 
Model Number: 96 of 133 with model FBProphet for Validation 2
96 - FBProphet with avg smape 39.01: 
Model Number: 97 of 133 with model SeasonalNaive for Validation 2
97 - SeasonalNaive with avg smape 15.6: 
Model Number: 98 of 133 with model GLS for Validation 2
98 - GLS with avg smape 12.78: 
Model Number: 99 of 133 with model GLS for Validation 2
99 - GLS with avg smape 12.78: 
Model Number: 100 of 133 with model FBProphet for Validation 2


INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


100 - FBProphet with avg smape 39.0: 
Model Number: 101 of 133 with model SeasonalNaive for Validation 2
101 - SeasonalNaive with avg smape 13.88: 
Model Number: 102 of 133 with model GLS for Validation 2
102 - GLS with avg smape 12.78: 
Model Number: 103 of 133 with model SeasonalNaive for Validation 2
103 - SeasonalNaive with avg smape 20.86: 
Model Number: 104 of 133 with model GLS for Validation 2
104 - GLS with avg smape 20.31: 
Model Number: 105 of 133 with model UnobservedComponents for Validation 2
105 - UnobservedComponents with avg smape 14.24: 
Model Number: 106 of 133 with model UnobservedComponents for Validation 2
106 - UnobservedComponents with avg smape 13.43: 
Model Number: 107 of 133 with model DatepartRegression for Validation 2
107 - DatepartRegression with avg smape 10.45: 
Model Number: 108 of 133 with model GLM for Validation 2
108 - GLM with avg smape 10.64: 
Model Number: 109 of 133 with model GLM for Validation 2
109 - GLM with avg smape 10.64: 
Model Number: 

/usr/local/lib/python3.7/dist-packages/sklearn/svm/_base.py:1208: ConvergenceWarning:

Liblinear failed to converge, increase the number of iterations.

[Parallel(n_jobs=-2)]: Using backend SequentialBackend with 1 concurrent workers.
[Parallel(n_jobs=-2)]: Done 1000 out of 1000 | elapsed:    4.1s finished
[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.
[Parallel(n_jobs=1)]: Done 1000 out of 1000 | elapsed:    0.3s finished


119 - DatepartRegression with avg smape 18.46: 
Model Number: 120 of 133 with model GLS for Validation 2
120 - GLS with avg smape 13.41: 
Model Number: 121 of 133 with model GLM for Validation 2
121 - GLM with avg smape 12.83: 
Model Number: 122 of 133 with model GLM for Validation 2
122 - GLM with avg smape 12.83: 
Model Number: 123 of 133 with model UnobservedComponents for Validation 2
123 - UnobservedComponents with avg smape 11.33: 
Model Number: 124 of 133 with model UnobservedComponents for Validation 2
124 - UnobservedComponents with avg smape 9.84: 
Model Number: 125 of 133 with model GLS for Validation 2
125 - GLS with avg smape 18.3: 
Model Number: 126 of 133 with model DatepartRegression for Validation 2


[Parallel(n_jobs=-2)]: Using backend SequentialBackend with 1 concurrent workers.
[Parallel(n_jobs=-2)]: Done 300 out of 300 | elapsed:    0.6s finished
[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.
[Parallel(n_jobs=1)]: Done 300 out of 300 | elapsed:    0.0s finished
[Parallel(n_jobs=-2)]: Using backend SequentialBackend with 1 concurrent workers.


126 - DatepartRegression with avg smape 15.22: 
Model Number: 127 of 133 with model DatepartRegression for Validation 2


[Parallel(n_jobs=-2)]: Done 100 out of 100 | elapsed:    0.4s finished
[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.
[Parallel(n_jobs=1)]: Done 100 out of 100 | elapsed:    0.0s finished


127 - DatepartRegression with avg smape 83.92: 
Model Number: 128 of 133 with model GLS for Validation 2
128 - GLS with avg smape 14.29: 
Model Number: 129 of 133 with model DatepartRegression for Validation 2
129 - DatepartRegression with avg smape 23.22: 
Model Number: 130 of 133 with model DatepartRegression for Validation 2
130 - DatepartRegression with avg smape 15.08: 
Model Number: 131 of 133 with model DatepartRegression for Validation 2
131 - DatepartRegression with avg smape 24.74: 
Model Number: 132 of 133 with model GLM for Validation 2
132 - GLM with avg smape 19.37: 
Model Number: 133 of 133 with model GLM for Validation 2
133 - GLM with avg smape 20.2: 


/usr/local/lib/python3.7/dist-packages/statsmodels/genmod/families/links.py:190: RuntimeWarning:

overflow encountered in exp

/usr/local/lib/python3.7/dist-packages/statsmodels/genmod/families/family.py:889: RuntimeWarning:

divide by zero encountered in true_divide



                   Close
2022-03-30  43124.201579
2022-03-31  43197.924577
2022-04-01  43192.725000
2022-04-02  43149.630908
2022-04-03  43051.157780
2022-04-04  42918.736117
2022-04-05  42828.810107
2022-04-06  42769.810596
2022-04-07  42734.819206
2022-04-08  42723.151676
2022-04-09  42750.008626
2022-04-10  42818.738493
2022-04-11  42912.449235
2022-04-12  43013.122949
2022-04-13  43138.881689
2022-04-14  43276.327653
2022-04-15  43417.107389
2022-04-16  43543.560872
2022-04-17  43627.030452
2022-04-18  43708.751302
2022-04-19  43766.532259
2022-04-20  43825.600814
2022-04-21  43885.929183
2022-04-22  43953.385677
2022-04-23  44001.035661
2022-04-24  43978.474837
2022-04-25  43907.937012
2022-04-26  43854.029639
2022-04-27  43831.032878
2022-04-28  43826.002930


**SUMMARY**

Buying and selling result in a change in the price of any cryptocurrency, but buying and selling trends depend on many factors. Using machine learning for cryptocurrency price prediction can only work in situations where prices change due to historical prices that people see before buying and selling their cryptocurrency. 